scripts/perftune.py

```diff
@@ -549,13 +563,27 @@ def _get_irqs(self):
         pass
 
 #### Private methods ############################
+    def __refresh_metadata_token(self):
+        self._metadata_token_time = datetime.datetime.now()
+        self._metadata_token = curl(self.EC2_METADATA_BASE_URL + "api/token", headers={"X-aws-ec2-metadata-token-ttl-seconds": self.EC2_METADATA_TOKEN_TTL}, method="PUT", timeout=0.1)
+
+    def __instance_metadata(self, path):
+        if not self._metadata_token:
+            self.__refresh_metadata_token()
+        else:
+            time_diff = datetime.datetime.now() - self._metadata_token_time
+            time_diff_sec = int(time_diff.total_seconds())
+            if time_diff_sec >= self.EC2_METADATA_TOKEN_TTL - 120:
+                self.__refresh_metadata_token()
+        return curl(self.EC2_METADATA_BASE_URL + path, headers={"X-aws-ec2-metadata-token": self._metadata_token}, timeout=0.1)
+
     def __check_host_type(self):
         """
         Check if we are running on the AWS i3 nonmetal instance.
         If yes, set self.__is_aws_i3_nonmetal_instance to True, and to False otherwise.
         """
         try:
-            aws_instance_type = urllib.request.urlopen("http://169.254.169.254/latest/meta-data/instance-type", timeout=0.1).read().decode()
+            aws_instance_type = self.__instance_metadata('meta-data/instance-type')
```

Comment:
Can it be anything other than `meta-data/...` for a metadata?

If you put this part of the URL inside the `def __instance_metadata(...)` as in my previous post the call will become:

```
aws_instance_type = self.__instance_metadata('instance-type')
```

Which is cleaner IMO.

Reply:
The other thing is `userdata/`, so it cannot be put `meta-data/` part into `__instance_metadata()` in scylla-machine-image (original code).
But in perftune.py we can just do it since we probably never reference userdata.

Fixed on the latest push.

Reply:
If we needed `userdata` you could have implemented a corresponding ,method: `__instance_userdata`.